pkg/tso/global_allocator.go

```diff
@@ -543,6 +543,11 @@ func (gta *GlobalTSOAllocator) campaignLeader() {
 	}()
 
 	gta.member.EnableLeader()
+	defer resetLeaderOnce.Do(func() {
```

Comment:
This seems to bring no difference. You added defer resetLeaderOnce.Do(... at the bottom of the code shown below, but it's already invoked so, if my understanding is corrected, what you added is duplicated and brings no difference. This piece of code refers to pd server.campaignLeader() in which "defer resetLeaderOnce.Do(..." is invoked twice, and I thought the second one is duplicate so I intentionally didn't invoke it after line 545 gta.member.EnableLeader(). Do I have misunderstanding? Maybe there is something that I should know but I don't know.

```
	ctx, cancel := context.WithCancel(gta.ctx)
	var resetLeaderOnce sync.Once
	defer resetLeaderOnce.Do(func() {
		cancel()
		gta.member.ResetLeader()
	})

	// maintain the the leadership, after this, TSO can be service.
	gta.member.KeepLeader(ctx)
        ...

	gta.member.EnableLeader()
	defer resetLeaderOnce.Do(func() {
		cancel()
		gta.member.ResetLeader()
	})
```

Reply:
As mentioned in the PR description, we need to make sure that ResetLeader happens before ResetAllocatorGroup.